### PR TITLE
ci: update CI workflows

### PR DESCRIPTION
**Pull request by maelle**
- removing the old check workflow -- this way you won't get all its failure notifications.
- trying to modernize the workflow that runs testother. It has been failing for a while: do you run these tests locally? or not at all? If not at all maybe the best way forward is to completely remove them?

### CODE REVIEW

File: DESCRIPTION [hunk 1]
```diff
@@ -51,3 +51,8 @@ RoxygenNote: 7.2.3
 X-schema.org-applicationCategory: Biodiversity
 X-schema.org-keywords: specimens, API, web-services, occurrences, species, taxonomy, GBIF, INAT, Vertnet, eBird, iDigBio, OBIS, ALA
 X-schema.org-isPartOf: ropensci.org
+Config/Needs/sftests:
```
Comment (maelle) [inline DESCRIPTION:54]: Listing the deps of the testother tests here, then the workflow installs them.

File: .github/workflows/test-sp-sf.yaml [hunk 2]
```diff
@@ -12,69 +18,31 @@ jobs:
...
-        shell: Rscript {0}
-
-      - name: Session info
-        run: |
-          options(width = 100)
-          pkgs <- installed.packages()[, "Package"]
-          sessioninfo::session_info(pkgs, include_base = TRUE)
-        shell: Rscript {0}
+          extra-packages: any::rcmdcheck
+          needs: check, sftests
```
Comment (maelle) [inline .github/workflows/test-sp-sf.yaml:45]: this is how the sftests deps are installed if I am not mistaken

**Comment by hannahlowens** [comment]
Hi Maëlle,
Thanks so much for this. I'm in favor of completely removing the testother workflow--I haven't been running it. Merging request.

Thanks again!

File: .github/workflows/test-sp-sf.yaml (hunk 2 above)
Comment (hannahlowens) [reply .github/workflows/test-sp-sf.yaml:41]: Makes sense to me.

**Review by hannahlowens — APPROVED** [review]
Looks good. Thank you.

**Pull request merged**